tests/components/RouterTest.php

```diff
@@ -36,7 +36,9 @@ public static function setUpBeforeClass():void {
          * @return void
          */
         protected function setUp():void {
-            $this->router = new Router();
+            $this->router = $this->getMockBuilder(Router::class)
+                                 ->onlyMethods(["invokeFactory"])
+                                 ->getMock();
```

Comment:
InvokeFactory method depends on factory classes and therefore it has to be mocked.